support-frontend/assets/pages/paper-subscription-checkout/helpers/paymentProviders.js

```diff
@@ -6,16 +6,76 @@ import {
   setupStripeCheckout,
 } from 'helpers/paymentIntegrations/stripeCheckout';
 import { type IsoCountry } from 'helpers/internationalisation/country';
-import type { PaymentAuthorisation } from 'helpers/paymentIntegrations/readerRevenueApis';
+import { type PaymentAuthorisation } from 'helpers/paymentIntegrations/readerRevenueApis';
 import {
   type PaymentResult,
+  type RegularPaymentRequest,
+  postRegularPaymentRequest, regularPaymentFieldsFromAuthorisation,
 } from 'helpers/paymentIntegrations/readerRevenueApis';
 import { type Dispatch } from 'redux';
 import { openDirectDebitPopUp } from 'components/directDebit/directDebitActions';
 import { finalPrice as paperFinalPrice } from 'helpers/productPrice/paperProductPrices';
+import { routes } from 'helpers/routes';
+import { getQueryParameter } from 'helpers/url';
+import { getOphanIds, getSupportAbTests } from 'helpers/tracking/acquisitions';
+import { Monthly } from 'helpers/billingPeriods';
+
 import { type State, setSubmissionError, setFormSubmitted, type Action, setStage } from '../paperSubscriptionCheckoutReducer';
 
-function onPaymentAuthorised(paymentAuthorisation: PaymentAuthorisation, dispatch: Dispatch<Action>) {
+function buildRegularPaymentRequest(state: State, paymentAuthorisation: PaymentAuthorisation): RegularPaymentRequest {
```

Comment:
I'm a bit concerned at this file being a copy of the one in `/digital-subscription-checkout`. It feels like we should make that file generic enough to handle both products and put it a top level file.

Reply:
@rupertbates I'd say for now let's copypaste them and before weekly (that's 3 already) let's work to identify the areas that are common enough to be extracted off. (Same story with the frontend bits such as the postcode finder)

Reply:
I see your concern is about this whole file @rupertbates , but I wanted to stop and think about what could be extracted specifically for this `buildRegularPaymentRequest` that is now in three places since this PR adds the third. But since it's roughly a mapper from the `CheckoutState` for each checkout to a `RegularPaymentRequest` I'm not sure how to do that without enforcing structure onto each `CheckoutState`.